brainframe_qt/api_utils/__init__.py

```diff
@@ -1,5 +1,38 @@
+import typing
+
+from PyQt5.QtCore import QObject
 from brainframe.api import BrainFrameAPI
-from .streaming import StreamManager, StreamManagerAPI
 
-# API instance that is later monkeypatched to be a singleton
-api: BrainFrameAPI = StreamManagerAPI()
+# Singleton API instance
+api = BrainFrameAPI()
+
+# Must come after api import
+from .streaming import StreamManager
+
+# Set using init_stream_manager before use
+#
+# Note: Creating singleton QObjects is a bit of a hassle because they need a parent
+# object. Flutter has a really cool system called InheritedWidget/Provider that makes
+# passing objects like this around really easily. I'd like to implement something
+# similar, but for now this is the only singleton object we have so I think this
+# should be begrudgingly "ok".
+_stream_manager: StreamManager = typing.cast(StreamManager, None)
+
+
+def init_stream_manager(*, parent: QObject) -> None:
+    global _stream_manager
+
+    if _stream_manager is not None:
+        raise RuntimeError("StreamManager has already been initialized")
+
+    _stream_manager = StreamManager(parent=parent)
```

Comment:
Are we okay with this not being thread-safe?

Reply:
It should be fine if its not thread safe. I think the worst that happens is that two stream managers get instantiated, one gets forgotten, and then GCd